spicy/runtime/include/configuration.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020-2021 by the Zeek Project. See LICENSE for details.
+
+#pragma once
+
+#include <functional>
+#include <optional>
+#include <string>
+
+namespace spicy::rt {
+
+/** Configuration parameters for the Spicy runtime system. */
+struct Configuration {
+    Configuration(){};
+
+    /**
+     * Optional callback to execute when a Spicy parser calls
+     * `spicy::confirm_input()`.
+     **/
+    std::optional<std::function<void()>> hook_confirm_input;
+
+    /**
+     * Optional callback to execute when a Spicy parser calls
+     * `spicy::reject_input()`. This string argument is the reason provided by
+     * the caller.
+     */
+    std::optional<std::function<void(const std::string&)>> hook_reject_input;
```

Comment:
WDYT about passing e.g., the global stream offset where this was called to the hooks? I am not exactly sure how this would be used by users, but a simple use case would be e.g., showing where the Spicy parser gave up?

The tricky bit would be in coming up with the offset which would not be passed directly by the user.

Reply:
I like that idea, but yeah, I don't see a way to access that information from where we call the hook, because it's just a function call and all we have are the user arguments. We could do it if we turned this into full Spicy statements, but not sure I like that. That said, the host application will have some idea where this is coming from, because it knows what data it has passed in most recently.